samples: net: sockets: echo:  update socket offload after net logging overhaul, update sample.yaml

GAnthony opened the pull request:
This patch gets the socket offload module and sockets echo sample to build after the net logging overhaul.

Also, adds updates sample.yaml to build for wifi based drivers (including SimpleLink).

Commit:
samples: net: include sockets echo sample build in cc3220sf sanitycheck

Previously, wifi-enabled samples were not built by sanitycheck for the
cc3220sf_launchxl.

This patch gets the sockets echo sample to build.

Signed-off-by: Gil Pitney <gil.pitney@linaro.org>

jukkar commented on samples/net/wifi/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+sample:
```
I did not notice that you had added sample.yaml file for wifi, so I have similar addition for that in #10405

GAnthony replied:
OK, then I'll remove this one.

pfalcon commented:
Commit message:

> wifi samples (sockets/echo, net/wifi)

Can you please replace with "wifi-enabled samples" to avoid confusion.

And you'll need to decide with @jukkar whose sample.yaml is better.

Commit:
net: sockets: Get socket offload to build after net logging overhaul.

Also, defines one LOG_MODULE_NAME for the simplelink WiFi driver, and
uses the same name for all files in this driver (module).

Signed-off-by: Gil Pitney <gil.pitney@linaro.org>

GAnthony commented:
OK, I removed my sample/net/wifi/sample.yaml, and also got socket offload and sockets/echo sample to build again with new logging updates.

jukkar commented on drivers/wifi/simplelink/simplelink_sockets.c:
```diff
@@ -4,6 +4,10 @@
 
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_socket_offload);
```
To my understanding, the module name must be unique but you are using the same name in socket_offload.c. Better to call this module "net_wifi_socket_offload".

GAnthony replied:
I noticed that each call to LOG_MODULE_REGISTER() is adding a new constant struct in memory for that file.
Reading logger.rst, and looking at how power module is using logging, it seemed a good opportunity to explore LOG_MODULE_DECLARE(), where the module name is registered only once, and other files which are part of the "module" only need to declare usage.
Here, I take the liberty of categorizing the "simplelink wifi driver" as a "module."
So, I'll push an update to this effect, and fix the shippable complaint on commit line length.

jukkar replied:
> Here, I take the liberty of categorizing the "simplelink wifi driver" as a "module."

Indeed, makes sense.

Commit:
drivers: wifi: simplelink: More fixes after logging update

Previously, '\n' in LOG_DBG statements were added to
improve readability, by ensuring IP address weren't printed
across typical console terminal line boundaries.

But with the logging updates, '\n' is no longer
getting automatically expanded to \r\n, and a new timestamp
is automatically added, throwing off attempts at alignment.

This patch finally just removes the '\n''s from the log messages,
and cuts a few longer messages into separate messages.

Signed-off-by: Gil Pitney <gil.pitney@linaro.org>

pfalcon commented on drivers/wifi/simplelink/simplelink_support.c:
```diff
@@ -218,7 +218,7 @@ void SimpleLinkWlanEventHandler(SlWlanEvent_t *wlan_event)
 		       BSSID_LEN_MAX);
 
-		LOG_INF("\n[WLAN EVENT] STA Connected to the AP: %s, "
+		LOG_INF("\n\r[WLAN EVENT] STA Connected to the AP: %s, "
```
I believe even when it was initially introduced, there was a comment that it's a bad thing, and it now pays off. So, it's another chance to remove that \n altogether, there's a simple rule to follow: one logging action - one log line produced.

If that doesn't work, then at lease make it \r\n, as the standard patter for it goes.

jukkar replied:
> \r\n, as the standard patter for it goes.

What standard is that? We should not add these \r, that is totally pointless thing to do. See also #10399 

I also think that adding \n to logging strings is useless as the output might go to some backend where \n makes no sense.

GAnthony replied:
OK, will (finally) remove those \n 's.

pfalcon replied:
:+1:

pfalcon reviewed (changes requested):
Needs some cleanup.

GAnthony commented:
OK, feedback resolved, and rebased.

pfalcon commented on drivers/wifi/simplelink/simplelink_log.h:
```diff
@@ -0,0 +1,23 @@
+/**
```
I don't remember this file in the previous PR version ;-). Let me approve, but IMHO, it's superfluous.

pfalcon replied:
2 lines of defines specifically. It does so using 23 lines. And it takes 1 line to include it. And it's included 3 times. For me, the math doesn't even out, but that's just me ;-).
